pandas/core/arrays/base.py

```diff
@@ -446,13 +446,24 @@ def shift(self, periods=1):
         Returns
         -------
         shifted : ExtensionArray
+
+        Notes
+        -----
+        If ``self`` is empty or ``periods`` is 0, a copy of ``self`` is
+        returned.
+
+        If ``periods > len(self)``, then an array of size
+        len(self) is returned, with all values filled with
+        ``self.dtype.na_value``.
         """
         # Note: this implementation assumes that `self.dtype.na_value` can be
         # stored in an instance of your ExtensionArray with `self.dtype`.
-        if periods == 0:
```

Comment:
what is len?

you mean len(self) right?

did the tests not catch this

Reply:
@jreback Shall I add fixtures for empty data at least for the 2 types addressed in this PR (`SparseArray` and `DecimalArray`)?

Reply:
you simply construct an empty from the data fixture

Reply:
specifically, `arr = data[:0]`

Reply:
yup got that.